tasks/src/tasks.rs

```diff
@@ -0,0 +1,75 @@
+//! Collection of common IMAP tasks.
+//!
+//! The tasks here correspond to the invocation (and processing) of a single command.
+
+use imap_codec::imap_types::{
+    command::CommandBody,
+    core::NonEmptyVec,
+    response::{Bye, Capability, Data, StatusBody, StatusKind},
+};
+
+use crate::Task;
+
+#[derive(Default)]
+pub struct CapabilityTask {
+    /// We use this as scratch space.
+    capabilities: Option<NonEmptyVec<Capability<'static>>>,
+}
+
+impl Task for CapabilityTask {
+    type Output = Result<NonEmptyVec<Capability<'static>>, &'static str>;
```

Comment:
Should `Result` be part of `Task`? I.e. `process_tagged` has the return type `Result<Self::Output, Self::Error>`.

Reply:
I'm not sure. I would propose we implement a few more `Task`s to gain a better understanding first.